File: Python/Adapter/Socket/electric_kettle.py
class ElectricKettle:
    def __init__(self, adapter) -> None:
        self.adapter = adapter

    def boil(self):
        if (self.adapter.voltage() > 110):
            print("Kettle on fire!")
        else:
            if (self.adapter.live() == 1 and self.adapter.neutral() == -1):
                print("Coffe time!")
            else:
                print("No adapter.")

File: JavaScript/Adapter/Socket/electricKettle.js
class ElectricKettle {
  constructor(power) {
    this._power = power;
  }

  boil() {
    if (this._power.voltage() > 110) {
      console.log("Kettle on fire!");
    } else {
      if (this._power.live() === 1 && this._power.neutral() === -1) {
        console.log("Coffee time!");
      } else {
        console.log("No power.");
      }
    }
  }
}

module.exports = ElectricKettle;
